exp/api/v1beta1/azuremachinepool_default_test.go

```diff
@@ -174,6 +202,45 @@ func TestAzureMachinePool_SetDiagnosticsDefaults(t *testing.T) {
 	g.Expect(nilDiagnostics.machinePool.Spec.Template.Diagnostics.Boot.StorageAccountType).To(Equal(infrav1.ManagedDiagnosticsStorage))
 }
 
+func TestAzureMachinePool_SetSpotEvictionPolicyDefaults(t *testing.T) {
+	g := NewWithT(t)
+
+	type test struct {
+		machinePool *AzureMachinePool
+	}
+
+	// test for SetSpotEvictionPolicyDefaults() to Ensure the default policy is being set correctly depending on the input1
```

Comment:
Apologies I should've been more clear in my last comment. I think it is worth re-writing this comment to be more descriptive. Maybe something like "Ensure the the default policy is set to Deallocate if diffDiskSettings is nil.